src/plugins/wizard/public/application/components/data_tab/secondary_panel.tsx

```diff
@@ -109,6 +109,20 @@ export function SecondaryPanel() {
           }}
           onAggTypeChange={function (aggId: string, aggType: IAggType): void {
             aggConfig.type = aggType;
+
+            // Persist field if the new agg type supports the existing field
+            const fieldParam = (aggType.params as AggParam[]).find(({ type }) => type === 'field');
```

Comment:
do we need the cast because the agg service doesn't give us suitable types?

Reply:
I'll be honest, that service has some seriously confusing typing going on that seems to confuse typescript a lot. Here the issue was that the line below where i try to `getAvailableFields` has a type error that only goes away when I cast these types. And these are valid types for those objects. Its just the unions and the parent child relationships between these types that is confusing the typescript compiler